siphon/catalog.py

```diff
@@ -670,14 +670,29 @@ def access_with_service(self, service):
         """
         service = CaseInsensitiveStr(service)
         if service == 'CdmRemote':
-            from .cdmr import Dataset as CDMRDataset
-            provider = CDMRDataset
+            if use_xarray:
+                from .cdmr.xarray_support import CDMRemoteStore
+                try:
+                    import xarray as xr
+                    provider = lambda url: xr.open_dataset(CDMRemoteStore(url))
+                except ImportError:
+                    raise ImportError('CdmRemote access needs xarray to be installed.')
+            else:
+                from .cdmr import Dataset as CDMRDataset
+                provider = CDMRDataset
         elif service == 'OPENDAP':
-            try:
-                from netCDF4 import Dataset as NC4Dataset
-                provider = NC4Dataset
-            except ImportError:
-                raise ImportError('OPENDAP access requires netCDF4-python to be installed.')
+            if use_xarray:
+                try:
+                    import xarray as xr
+                    provider = xr.open_dataset
+                except ImportError:
+                    raise ImportError('OPENDAP access needs xarray to be installed.')
```

Comment:
Should this be OPENDAP needs xarray or something specific to using the xarray accessor like 'xarray must be installed to use xarray access method.' ?

Reply:
Sure. Fixed.